docs/marketplace/dev/front-end/table-design/table-column-types/table-column-type-select.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Table Column Type Select
+description: This document provides details about the Table Column Type Select in the Components Library.
+template: concept-topic-template
+---
+
+This document explains the Table Column Type Select in the Components library.
+
+## Overview
+
+Table Column Select is an Angular Component that renders a drop-down list using the `@spryker/select` component.
+
+Example usage of the Table Column Select in the `@spryker/table` config:
+
+```html
+<spy-table [config]="{
+    ...,
+    columns: [
+      ...
+      {
+        id: 'columnId',
+        title: 'Column Title',
+        type: 'select',
+        typeOptions: {
+          options: [
+            {
+              title: 'Option 1',
+              value: 'Option 1',
+            },
+            {
+              title: 'Option 2',
+              value: 'Option 2',
+              isDisabled: true,
+            },
+            {
+              title: 'Option 3',
+              value: 'Option 3',
+            },
+          ],
+          placeholder: '123',
+        },
+      },
+      ...
+    ]
+  }"
+>
+</spy-table>
+```
+
+## Interfaces
+
+Below you can find interfaces for the Table Column Select type:
+
+```ts
+type SelectValue = string | number;
+type SelectOption = SelectValue | SelectOptionItem;
+
+interface SelectOptionItem {
+    title: string;
+    value: SelectValue;
+    isDisabled?: boolean;
+}
+
+interface DataTransformerConfig {
+    type: string;
+    // Reserved for types that may have extra configuration
+    [extraConfig: string]: unknown;
+}
+
+interface DatasourceConfig {
+    type: string;
+    transform?: DataTransformerConfig;
+    // Specific Datasource types may have custom props
+    [k: string]: unknown;
+}
```

Comment:
Why do we need these types here?

Reply:
Because `ColumnSelectDatasource` and `ColumnSelectDataTransformer` implement them.